dash/development/base_component.py

```diff
@@ -87,6 +100,40 @@ def __init__(self, **kwargs):
                         ', '.join(sorted(self._prop_names))
                     )
                 )
+
+        # Make sure arguments have valid values
+        self._validator = cerberus.Validator(self._schema, allow_unknown=True)
+        valid = self._validator.validate(kwargs)
+        if not valid:
+            raise TypeError("""
+                Invalid initialization of component {}.
+                The errors are as follows:
+                Errors:
+
+                {}
+            """.format(
+                self.__class__.__name__,
+                "\n\n".join([(
+                    "Invalid value {} for property {}.\n"
+                    "This broke the rule {} which should be {}\n"
+                    "Here is the expected schema for the {} prop:\n\n"
+                    "{}"
+                )
+                 .format(
+                     e.value,
+                     e.schema_path[0],
+                     e.schema_path,
+                     e.constraint,
+                     e.schema_path[0],
+                     json.dumps(
+                         self._schema[e.schema_path[0]],
+                         indent=1
+                     ))
```

Comment:
I've had a really hard time getting useful information from deeply nested errors with cerberus. Have you had better luck?

Reply:
Errors coming from nested schemas seem to only give error information about the top schema (e.g. the second example given above), I think we may be able to get something better by digging into the `ValidationError` class (http://docs.python-cerberus.org/en/stable/errors.html#python-interfaces). I'll take a look and see what we can do.